apollo-federation/src/link/federation_spec_definition.rs

```diff
@@ -1065,3 +1088,249 @@ pub(crate) fn fed1_link_imports() -> Vec<Arc<link::Import>> {
         .map(Arc::new)
         .collect()
 }
+
+pub fn get_federation_spec_definition_string_from_version(
+    version: Version,
+) -> Result<String, FederationError> {
+    let fed_spec_definition = FEDERATION_VERSIONS.find(&version).ok_or_else(|| {
+        SingleFederationError::UnknownFederationLinkVersion {
+            message: format!("Unknown federation spec version: {}", version),
+        }
+    })?;
```

Comment:
You can use `let-else` here:
```suggestion
    let fed_spec_definition = FEDERATION_VERSIONS.find(&version) else {
        return SingleFederationError::UnknownFederationLinkVersion {
            message: format!("Unknown federation spec version: {}", version),
        }.into()
    };
```
In most of the codebase, this pattern of `Option -> Result` via `ok_or_else + ?` was replaced with let-else (and `bail!` where useable).

Reply:
<img width="518" alt="image" src="https://github.com/user-attachments/assets/836e92b0-e8b6-4faf-b914-076abb783286" />

I'm good with changing this to match the codebase's pattern, but I do get an error about the else clause being superfluous. The suggested fix for that brings me back to `ok_or_else()`.

For context, this was the full change I made:
```suggestion
    let fed_spec_definition = FEDERATION_VERSIONS.find(&version) else {
        return Err(SingleFederationError::UnknownFederationLinkVersion {
            message: format!("Unknown federation spec version: {}", version),
        }.into());
    };

    fed_spec_definition.expect("Federation spec definition should exist after version check").federation_spec_to_string()
```